libheif/context.cc

```diff
@@ -517,6 +517,18 @@ static bool item_type_is_image(const std::string& item_type, const std::string&
           item_type == "mski");
 }
 
+#if WITH_EXPERIMENTAL_GAIN_MAP
+static bool item_type_is_gain_map_image(const std::string& item_type, const std::string& item_name) {
+  return  (item_name == "GMap" &&
```

Comment:
It appears that images are not using a `GMap` `item_name`. (see my comment in the discussion.)